### PR TITLE
Secrets Sync

### PR DESCRIPTION
**Pull request by zofskeez**
This PR adds support for secrets syncing in the UI. The new sync enterprise feature enables users to sync KVv2 secrets in Vault to external destinations such as AWS Secrets Manager, Azure Key Vault and more.

https://github.com/hashicorp/vault/assets/24611656/4bc53627-19fb-40af-9db9-57b74d2604a9

### CODE REVIEW

File: ui/app/models/sync/destinations/aws-sm.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+export default class SyncDestinationsAwsSecretsManagerModel extends SyncDestinationModel {
+  @attr('string', {
+    label: 'Access key ID',
+    subText:
+      'Access key ID to authenticate against the secrets manager. If empty, Vault will use the AWS_ACCESS_KEY_ID environment variable if configured.',
+  })
+  accessKeyId; // obfuscated, never returned by API
+
+  @attr('string', {
+    label: 'Secret access key',
```
Comment (hashishaw) [inline ui/app/models/sync/destinations/aws-sm.js:25]: Should this and accessKeyId be masked inputs?

File: ui/app/models/sync/destinations/azure-kv.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    subText:
+      'Client ID of an Azure app registration. If empty, Vault will use the AZURE_CLIENT_ID environment variable if configured.',
+  })
+  clientId;
+
+  @attr('string', {
+    subText:
+      'Client secret of an Azure app registration. If empty, Vault will use the AZURE_CLIENT_SECRET environment variable if configured.',
+  })
+  clientSecret; // obfuscated, never returned by API
```
Comment (hashishaw) [inline ui/app/models/sync/destinations/azure-kv.js:35]: Same with this. Really any secrets should be masked, I would think.

File: ui/app/models/sync/destinations/gcp-sm.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  @attr('string', {
+    label: 'JSON credentials',
+    subText:
+      'If empty, Vault will use the GOOGLE_APPLICATION_CREDENTIALS environment variable if configured.',
+    editType: 'file',
+    docLink: '/vault/docs/secrets/gcp#authentication',
+  })
+  credentials; // obfuscated, never returned by API
+
+  // TODO - confirm if project_id is going to be added to READ response (not editable)
```
Comment (hashishaw) [inline ui/app/models/sync/destinations/gcp-sm.js:23]: 🧹

File: ui/lib/sync/addon/components/secrets/page/overview.hbs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      <Hds::Button @text="Create first destination" @route="secrets.destinations.create" data-test-cta-button />
+    {{/unless}}
+  </:actions>
+</SyncHeader>
+
+{{#if @destinations}}
+  <div class="tabs-container box is-bottomless is-marginless is-paddingless">
+    <nav class="tabs" aria-label="destination tabs">
+      <ul>
+        <LinkTo @route="secrets.overview" data-test-tab="Overview">Overview</LinkTo>
```
Comment (hashishaw) [inline ui/lib/sync/addon/components/secrets/page/overview.hbs:18]: These need to be wrapped in `<li>`
```suggestion
        <li><LinkTo @route="secrets.overview" data-test-tab="Overview">Overview</LinkTo></li>
```

File: ui/lib/sync/addon/components/secrets/page/overview.hbs [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        <LinkTo @route="secrets.destinations" data-test-tab="Destinations">Destinations</LinkTo>
```
Comment (hashishaw) [inline ui/lib/sync/addon/components/secrets/page/overview.hbs:19]:
```suggestion
        <li><LinkTo @route="secrets.destinations" data-test-tab="Destinations">Destinations</LinkTo></li>
```

File: ui/lib/sync/addon/components/secrets/page/overview.hbs [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+      </ul>
+    </nav>
+  </div>
+  <Toolbar>
```
Comment (hashishaw) [inline ui/lib/sync/addon/components/secrets/page/overview.hbs:23]: This, ToolbarActions, and ToolbarLink needs `aria-label`

File: ui/tests/helpers/general-selectors.js
Comment (hashishaw) [inline ui/tests/helpers/general-selectors.js:1]: 🎉

**Review by hashishaw — APPROVED** [review]
A couple minor nits and a11y change requests, but otherwise looks great!

File: ui/app/models/sync/destinations/aws-sm.js (hunk 1 above)
Comment (hellobontempo) [reply ui/app/models/sync/destinations/aws-sm.js:25]: They weren't in the designs 🤔 But this is a good question - we have a meeting scheduled to do a design review/walk through in the new year. I'll make a note to ask this!

File: ui/app/models/sync/destinations/azure-kv.js (hunk 2 above)
Comment (hellobontempo) [reply ui/app/models/sync/destinations/azure-kv.js:35]: We don't mask them for OIDC config...but maybe that was built before we used that component

File: ui/app/models/sync/destinations/gcp-sm.js (hunk 3 above)
Comment (hellobontempo) [reply ui/app/models/sync/destinations/gcp-sm.js:23]: This is still a TODO 😞 - waiting for manually testing creds to confirm this

File: ui/lib/sync/addon/components/secrets/page/overview.hbs (hunk 4 above)
Comment (hellobontempo) [reply ui/lib/sync/addon/components/secrets/page/overview.hbs:18]: good catch!

**Pull request merged**